EventFilter/EcalRawToDigi/plugins/alpaka/DeclsForKernels.h

```diff
@@ -0,0 +1,43 @@
+#ifndef EventFilter_EcalRawToDigi_plugins_alpaka_DeclsForKernels_h
+#define EventFilter_EcalRawToDigi_plugins_alpaka_DeclsForKernels_h
+
+#include "HeterogeneousCore/AlpakaInterface/interface/memory.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/config.h"
+
+namespace ALPAKA_ACCELERATOR_NAMESPACE::ecal::raw {
+
+  struct InputDataHost {
+    // delete the default constructor because alpaka buffers do not have a default constructor
+    InputDataHost() = delete;
+
+    explicit InputDataHost(const Queue& queue, size_t size, size_t nFeds)
+        : data{cms::alpakatools::make_host_buffer<unsigned char[]>(queue, size)},
+          offsets{cms::alpakatools::make_host_buffer<uint32_t[]>(queue, nFeds)},
+          feds{cms::alpakatools::make_host_buffer<int[]>(queue, nFeds)} {};
+
+    cms::alpakatools::host_buffer<unsigned char[]> data;
+    cms::alpakatools::host_buffer<uint32_t[]> offsets;
+    cms::alpakatools::host_buffer<int[]> feds;
+  };
+
+  struct ConfigurationParameters {
+    uint32_t maxChannelsEE;
+    uint32_t maxChannelsEB;
+  };
```

Comment:
`ConfigurationParameters` could be moved out of `ALPAKA_ACCELERATOR_NAMESPACE` into `namespace ecal::raw`.

Reply:
There is actually a version of this struct  already in that namespace. It is the one for the CUDA implementation [DeclsForKernels.h](https://github.com/cms-sw/cmssw/blob/master/EventFilter/EcalRawToDigi/plugins/DeclsForKernels.h#L24).
In order to avoid mixing CUDA and alpaka code the struct was duplicated inside `ALPAKA_ACCELERATOR_NAMESPACE`.

Reply:
Mhm, I see.
For a more complex type, I'd suggest sharing it between the CUDA and Alpaka code base to reduce duplication - but this is simple enough that duplicating it is not much extra code.

Maybe it can be moved out of the `ALPAKA_ACCELERATOR_NAMESPACE` after the CUDA code is removed.

Reply:
Yes I would move it out of the namespace once the CUDA code is removed.